cs/client.py

```diff
@@ -73,6 +74,20 @@
 FAILURE = 2
 
 
+def check_key(key, allowed):
+    """
+    Validate that the specified key is allowed according the provided
+    list of patterns.
+    """
+
+    from fnmatch import fnmatch
+    for pattern in allowed:
```

Comment:
You could short-circuit the `for` loop to optimize the exact match case:

```python
if pattern in allowed:
    return True

for pattern in allowed:
    if fnmatch(…)
```

Reply:
Thanks, fixed.